### ISSUE TITLE
Harden dashboard addon

### ISSUE THREAD

**Issue by pjbgf** [kubernetes/kubernetes#82647]
**What would you like to be added**:
Apply the following security best practices for dashboard add-on:

Low hanging fruit:

- [x]  Run as non-root (& disallow privilege escalation)
- [x] Run with the default seccomp profile (i.e.  `runtime/default`)
- [x] ReadOnlyRootFilesystem
- [x] Avoid unnecessary HostPath volumes
- [x]  ~~Don't mount service account token~~ (n/a)

More advanced:

- [x] Reduce dependencies in the base image (#40248)
- [ ] Custom seccomp profile
- [ ] Custom AppArmor profile
- [x] ~~Drop unneeded capabilities~~ (n/a: running as non-root)

This is based on the long term issue #38541.

**Why is this needed**:
Our system pods should run using security best practices, both to enhance cluster security and serve as examples of best practices to users.

/cc @tallclair 
/sig auth
/area security

**Comment by pjbgf** [comment]
Create a new PR (https://github.com/kubernetes/dashboard/pull/4288) to resolve: 
- Reduce dependencies in the base image

Once accepted, merged and a new image published, we can tick that one off.

**Comment by tallclair** [comment]
BTW, I consider custom AppArmor / Seccomp profiles to be pretty low priority, especially as we don't have a good story for deploying custom profiles right now. Also matters less for non-root containers.

**Comment by pjbgf** [comment]
@tallclair custom seccomp profiles may change depending on the value of `securityContext.allowPrivilegeEscalation`. I propose that we only support a single profile version assuming (and recommending that) users should always deploy this component using `allowPrivilegeEscalation=false` (i.e. `--no-new-privileges`). This decreases the risk of breaking the profile due to external changes (e.g. runc, docker, etc).

**Comment by pjbgf** [comment]
The seccomp custom profile is blocked by https://github.com/kubernetes/kubernetes/issues/84623.

**Comment by pjbgf** [comment]
Still relevant. The seccomp profile is waiting for https://github.com/kubernetes/kubernetes/pull/85121 to be approved.

/remove-lifecycle stale

**Comment by pjbgf** [comment]
A least privileged custom seccomp profile for the dashboard depends on a few PRs being merged (https://github.com/kubernetes/kubernetes/pull/90949, https://github.com/kubernetes/kubernetes/pull/90948), as pointed out on issue #84623.

**Comment by pjbgf** [comment]
remove-lifecycle stale

**Comment by pjbgf** [comment]
Ticking "Reduce dependencies in the base image" off as the dashboard is based off scratch for a while now.

**Comment by tallclair** [comment]
I think we can close this. The remaining tasks are pretty advanced hardening strategies with debatable value at this point. Any further hardening issues should be opened against the dashboard repo: https://github.com/kubernetes/dashboard/issues
